src/main/java/com/amazon/opendistroforelasticsearch/knn/plugin/script/KNNScoringUtil.java

```diff
@@ -13,6 +13,25 @@ public static float l2Squared(float[] queryVector, float[] inputVector) {
         return squaredDistance;
     }
 
+    public static float cosinesimilOptimized(float[] queryVector, float[] inputVector, double normQueryVector) {
+        double dotProduct = 0.0f;
+        double normInputVector = 0.0f;
```

Comment:
since input and outputs are all float, why not using float?

Reply:
note that every casting, even implicit, takes away resource and gives no gains

Reply:
Just to add on, I ran some benchmarking on a single i3.4xl of a tight dot product loop, and found the casting back and forth to double slowed down the calculation measurably.

Reply:
Good point. Fixed